shared/actions/chat2/index.tsx

```diff
@@ -662,12 +701,40 @@ const onChatSetConvRetention = (_, action, logger) => {
   })
 }
 
-const onChatSetConvSettings = (_, action, logger) => {
+const onChatSetConvSettings = (_, action: EngineGen.Chat1NotifyChatChatSetConvSettingsPayload, logger) => {
   const {conv, convID} = action.payload.params
   const conversationIDKey = Types.conversationIDToKey(convID)
-  const newRole = conv?.convSettings?.minWriterRoleInfo?.role
+  const newRole = // Auto generated from flowToTs. Please clean me!
```

Comment:
i think we need to fix this

Reply:
I changed it to   `const cannotWrite = conv && conv.convSettings && conv.convSettings.minWriterRoleInfo && conv.convSettings.minWriterRoleInfo.cannotWrite || null` which is technically different than the nullish coalescing, but fine for this I think.